### ISSUE TITLE
'wait_until' method with 'text:' option doesn't work correctly without specifying timeout value

### ISSUE THREAD

**Issue by nbulavin** [natritmeyer/site_prism#299]
## Environment
- **Operating System:** Windows
- **What Gems / Versions you are using:** site_prism 2.14, Capybara 2.18.0
- **How you're interacting with the Framework:** Cucumber Tests
- **Browser/s Driver/s being used:** Chrome 68.0.3440.75, ChromeDriver v.2.40, Selenium 3.11.0

## Expected Behavior
'wait_until_<element_or_section_name>_visible' method works with 'text:' option without specifying timeout value as described in documentation here:
https://github.com/natritmeyer/site_prism#methods-supporting-capybara-options
`@results_page.wait_until_<element_or_section_name>_visible(text: 'Some ajaxy text appears!')`

## Actual Behavior
Method ingores `text:` param (as I think) until I set timeout param explicitly

## Steps to reproduce
Capybara config:
```
Capybara.configure do |c|
  c.default_selector = :xpath
  c.default_max_wait_time = 30
  c.ignore_hidden_elements = true
end
```
Siteprism config:
```
SitePrism.configure do |c|
  c.raise_on_wait_fors = true
end
```
Page class example:
```
class Github < SitePrism::Page
  set_url 'https://github.com/'

  element :login_link, '//a[@href="/login"]'
end
```
Cucumber step example:
```
When(/^I look at login button$/) do
  @github.load
  @github.wait_until_login_link_visible(text: 'any text')
end
```
When I use only `text:` option then step finished succesfully

If I set timeout:
`@github.wait_until_login_link_visible(10, text: 'any text')`
then step is failed on this method as expected

**Comment by nbulavin** [comment]
Removed options and issue still reproducible. 

Attached [for_site_prism.zip](https://github.com/natritmeyer/site_prism/files/2242135/for_site_prism.zip)

**Comment by luke-hill** [comment]
Cheers. Will triage ASAP

**Comment by luke-hill** [comment]
Ok so the short answer, is because that portion of the README was written in This commit https://github.com/natritmeyer/site_prism/commit/3ba7944a104804f825fc9e1f48d4bda785a0a7a9 (Which is ancient), I don't believe the way the API was designed will facilitate what you need.

The quick fix is to change your code to `@github.wait_until_login_link_visible(nil, text: 'any text')` -> Note that this is actually mentioned in some code comments in the README above for the `wait_for` methods.

I think in essence you've spotted a slight flaw (And in internal test documents, we cover this), where our API design sets the `timeout` separately to every other runtime argument. And in essence *it shouldn't*

I've added this to the roadmap for v3.0 as this needs fixing, but this will break every single metaprogrammed method from going from.

`wait_for_element(10)` to `wait_for_element(wait: 10)`

With the latter being more desirable due to being hashable and passable as a query.

Once we've made the minor (But breaking change), mentioned. You will be able to do your code correctly.

Now thats the good news......

The bad news....

We need to get a minimum of 4 more code releases out of features / tweaks I want to make, as well as get some more of the backlog of tasks completed.

I also need to chat to the original author as this would be a reasonably large change to the docs and methodology.

So an ETA for the changes to make your code work as you expect is a long way off sadly.

TL;DR - add `nil` as your first parameter. Or you can use the `has_?` methods which already work in this way (And facilitate waits via implicit waiting)

**Comment by luke-hill** [comment]
@nbulavin - This is fixed in the current master branch and will be cut as an alpha 3.0 release (Alongside the other items in master), sometime in the coming weeks (Probably around 25th-30th)
